api/src/main/java/org/openmrs/module/databasebackup/util/DbDump.java

```diff
@@ -183,42 +183,83 @@ public static void dumpDB(Properties props, boolean showProgress, Class showProg
+            r.next();
+            int count = r.getInt("rowcount");
+            r.close();
+
+            // Variables for dividing rows
+
+            int startIndex = 0;
+            boolean notProcessed = true;
+            int offset = 0;
+
+            result.write("\n\n-- Data for table '" + tableName + "'\n");
+
+            while (notProcessed) {
+
+                PreparedStatement stmt = dbConn.prepareStatement("SELECT * FROM " + tableName + " LIMIT " + String.valueOf(offset) + ", " + String.valueOf(max) + ";");
+                ResultSet rs = stmt.executeQuery();
+                ResultSetMetaData metaData = rs.getMetaData();
+                int columnCount = metaData.getColumnCount();
+
+                String dataHeaders = "(" + metaData.getColumnName(1);
+                for (int i = 2; i <= columnCount; i++) {
+                    dataHeaders += "," + metaData.getColumnName(i);
+                }
+                dataHeaders += ")";
+
+                // Check that row exist before insert statement can be created
+                if(rs.isBeforeFirst()) {
+                    result.write("INSERT INTO " + tableName + " " + dataHeaders + " VALUES ");
+                }
+                // data inserts
+                while (rs.next()) {
+
+                    result.write("(");
+                    for (int i = 0; i < columnCount; i++) {
+                        if (i > 0) {
+                            result.write(", ");
+                        }
+                        Object value = rs.getObject(i + 1);
+                        if (value == null) {
+                            result.write("NULL");
+                        } else {
+                            String outputValue = value.toString();
+                            if (value instanceof Boolean) {
+                                outputValue = (((Boolean) value) ? "1" : "0");
+                            }
+                            outputValue = escape(outputValue);
+                            result.write("'" + outputValue + "'");
+                        }
                     }
-                    Object value = rs.getObject(i+1);
-                    if (value == null) {
-                        result.write("NULL");
+                    if (rs.isLast()) {
+                        result.write(");\n");
                     } else {
-                        String outputValue = value.toString();
-                        if (value instanceof Boolean) {
-                        	outputValue = (((Boolean)value) ? "1" : "0");
-                        }
-                        outputValue = escape(outputValue);
-                        result.write( "'"+outputValue+"'" );
+                        result.write("),");
                     }
                 }
-                result.write(");\n");
+                rs.close();
+                stmt.close();
+
+                if (offset >= count || count <= max) {
```

Comment:
The first loop breaks if all the rows have been processed, the offset is greater than total number of rows in the table. I am not getting exactly why you say just break out of the loop